app/models/casebook/resource.rb

```diff
@@ -0,0 +1,32 @@
+# == Schema Information
+#
+# Table name: casebooks
+#
+#  id            :integer          not null, primary key
+#  title         :string           default("Untitled casebook"), not null
+#  slug          :string
+#  subtitle      :string
+#  headnote      :text
+#  public        :boolean          default(TRUE), not null
+#  root_id       :integer
+#  ordinals      :integer          default([]), not null, is an Array
+#  copy_of_id    :integer
+#  is_alias      :boolean
+#  material_type :string
+#  material_id   :integer
+#  created_at    :datetime         not null
+#  updated_at    :datetime         not null
+#
+
+class Casebook::Resource < Casebook::Contents
+  default_scope {where.not(material_id: nil)}
+
+  belongs_to :material, polymorphic: true, inverse_of: :casebooks, required: true
```

Comment:
(This is what the `is_alias` boolean column is theoretically for. If `is_alias` is true, this Resource record points to a Case (for example) that is shared by another casebook, and it should be copied before making edits.)